### PR TITLE
feat(ONYX-651): Implement tracking events for NWFY

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-651] 

### Description

This PR makes the follow changes:
- Add tracking to NWFY screen
- Add a dev toggle to force seeing the feed variant

**Note:** I created this branch from this [one](https://github.com/artsy/eigen/pull/9724) - this is why the big changes, they should be solved as soon as the other gets merged

https://github.com/artsy/eigen/assets/11945712/10a8ce5f-c66e-45e9-88bb-98c24e3111d3

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Implement tracking events for NWFY - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-651]: https://artsyproduct.atlassian.net/browse/ONYX-651?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Implement tracking events for NWFY - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c49c10e2b15eca1ba90dc761b07c60316a31c0d1
</p>

File: src/app/Scenes/NewWorksForYou/NewWorksForYou.tsx [hunk 1]
```diff
@@ -63,16 +67,27 @@ export const NewWorksForYouQueryRenderer: React.FC<NewWorksForYouQueryRendererPr
...
+      experiment.enabled &&
+      (experiment.variant === "experiment" || forceShowNewWorksForYouFeed)
+    ) {
+      return true
+    }
+
+    return false
+  }
+
+  if (showFeed()) {
```
Comment (olerichter00) [inline src/app/Scenes/NewWorksForYou/NewWorksForYou.tsx:90]: **suggestion:** Less code and maybe easier to read 🤔 
```suggestion
  const showFeed = !isTablet() && enableNewWorksForYouFeed && experiment.enabled && (experiment.variant === "experiment" || forceShowNewWorksForYouFeed)

  if (showFeed) {
```

**Review by olerichter00 — DISMISSED** [review]
Looks good 🌟

File: src/app/Scenes/NewWorksForYou/NewWorksForYou.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/NewWorksForYou/NewWorksForYou.tsx:90]: sure, I actually find the other version more readable but will gladly update this

**Pull request merged**